editor/editor_properties_array_dict.cpp

```diff
@@ -1177,7 +1177,7 @@ void EditorPropertyDictionary::update_property() {
 
 			// We need to grab the focus of the property that is being changed, even if the type didn't actually changed.
 			// Otherwise, focus will stay on the change type button, which is not very user friendly.
-			if (changing_type_index == slot.index) {
+			if (changing_type_index == slot.index && slot.index != EditorPropertyDictionaryObject::NEW_KEY_INDEX && slot.index != EditorPropertyDictionaryObject::NEW_VALUE_INDEX) {
```

Comment:
This, while fixing the problem, looks strange, why can the `slot.index` be negative? It shouldn't be right?

Reply:
> This, while fixing the problem, looks strange, why can the `slot.index` be negative? It shouldn't be right?
> 
> Edit: Realized that it does in fact handle the special slots that way, so this is correct, but can be simplified like so I'd say:
> 
> Like is done below

It's definitely simpler and shorter, but it's hard to understand the motive IMO. I would rather leave the more expressive, longer one.

If we are to simplify it anyway, I suggest adding a comment explaining why.

Reply:
It's used everywhere else we do the same check, like on line 899, 946, 964, 1152, so I'd say it's pretty established as the method to indicate that

Reply:
We could do this too:
```
bool EditorPropertyDictionary::_is_slot_index_new_key_or_new_value(int i) {
	return i > -1;
}

if (changing_type_index == slot.index && !_is_slot_index_new_key_or_new_value(slot.index)) {
```

Reply:
Note that it should *never* change so that the first valid element index is `0`, that wouldn't make sense, it should always be valid that "normal" entries are `slot.index > -1`

Reply:
Would be worth clarifying why we check that in general though, but unrelated to the specific choice of `> -1` or the explicit check